packages/keyboard-shortcuts/src/store/actions.js

```diff
@@ -1,22 +1,24 @@
+/** @typedef {import('@wordpress/keycodes').WPKeycodeModifier} WPKeycodeModifier */
+
 /**
  * Keyboard key combination.
  *
  * @typedef {Object} WPShortcutKeyCombination
  *
- * @property {string}  character Character.
- * @property {string} [modifier]  Modifier.
+ * @property {string}                      character Character.
+ * @property {WPKeycodeModifier|undefined} modifier  Modifier.
```

Comment:
This change was a bit unfortunate, I found. The previous `[modifier]` optional property syntax should be fine, but for some reason the TypeScript tooling does not pick up on it being an issue unless it was explicitly a union type with `undefined`, even though it detects it as optional.

It only (correctly) flags an error when trying to use a modifier method on `WPKeycodeHandlerByModifier` when it's typed using the undefined union:

![undefined-union](https://user-images.githubusercontent.com/1779930/71729460-c8690600-2e0d-11ea-9585-8ce2ffdc3686.gif)